test/integration/graql/query/GraqlDeleteIT.java

```diff
@@ -110,527 +110,8 @@ public static void closeSession() {
-                        .rel("refl", "p")
-                        .rel("refl", "p")
-                        .rel("refl", "p"),
-                var("p").isa("athing"))
-        );
-
-        List<ConceptMap> answers = tx.execute(Graql.match(var("r").isa("reflexive").rel("refl", "x").rel("refl", "x")).get());
-        assertEquals(1, answers.size());
-        assertEquals(3, Iterators.getOnlyElement(answers.get(0).get("r").asRelation().rolePlayersMap().values().iterator()).size());
-
-        tx.execute(Graql.match(var("r").isa("reflexive").rel("refl", "x").rel("refl", "x"))
-                .delete(var("r").rel("refl", "x").rel("refl", "x")));
-
-        answers = tx.execute(Graql.match(var("r").isa("reflexive").rel("refl", "x")).get());
-        assertEquals(1, answers.size());
-        assertEquals(1, Iterators.getOnlyElement(answers.get(0).get("r").asRelation().rolePlayersMap().values().iterator()).size());
-    }
-
-    @Test
-    public void whenDeletingTooManyDuplicateRolePlayers_Throw() {
-        tx.execute(Graql.parse("define reflexive sub relation, relates refl; athing sub entity, plays refl;").asDefine());
-        tx.execute(Graql.insert(
-                var("r")
-                        .isa("reflexive")
-                        .rel("refl", "p"),
-                var("p").isa("athing"))
-        );
-
-        List<ConceptMap> answers = tx.execute(Graql.match(var("r").isa("reflexive").rel("refl", "x")).get());
-        assertEquals(1, answers.size());
-        assertEquals(1, Iterators.getOnlyElement(answers.get(0).get("r").asRelation().rolePlayersMap().values().iterator()).size());
-
-        exception.expect(GraqlSemanticException.class);
-        exception.expectMessage("Cannot delete role player [$x]");
-        exception.expectMessage("it does not play required role (or subtypes of) [refl]");
-        // match it once, delete it as a duplicate but we only have a single player! Should throw
-        tx.execute(Graql.match(var("r").isa("reflexive").rel("refl", "x"))
-                .delete(var("r").rel("refl", "x").rel("refl", "x")));
-    }
-
-    /*
-    We allow the following generalised role in the delete clause:
-    match $r (sub-role: $x) isa relation; delete $r (super-role: $x);
-     */
-    @Test
-    public void whenDeletingRolePlayerAsSuperRole_PlayerIsRemoved() {
-        tx.execute(Graql.parse("match $r (production-being-directed: $x, director: $y) isa directed-by; delete $r (role: $x);").asDelete());
-        List<ConceptMap> answers = tx.execute(Graql.parse("match $r (director: $y) isa directed-by; get;").asGet());
-        assertEquals(1, answers.size());
-        assertEquals(1, answers.get(0).get("r").asRelation().rolePlayers().count());
-    }
-
-    /*
-    We don't allow the following downcasted role in the delete clause:
-    match $r (super-role: $x) isa relation; delete $r (sub-role: $x);
-     */
-    @Test
-    public void whenDeletingRolePlayerAsSubRole_Throw() {
```

Comment:
double check this has been migrated, only recall seeing that we don't throw when using the super-role

Reply:
Again this scenario was in draft and I forgot to implement it. It has now been implemented!! You have a keen eye.